torch/testing/_internal/common_utils.py

```diff
@@ -1370,31 +1374,53 @@ def settings(*args, **kwargs):
 except ImportError:
     print('Fail to import hypothesis in common_utils, tests are not derandomized')
 
+
+def remove_device_and_dtype_suffixes(test_name: str) -> str:
```

Comment:
Add a comment explaining what this function does and how it's intended to be used